blockchain/filedao/filedao_v2_util.go

```diff
@@ -178,12 +173,7 @@ func (fd *fileDAOv2) getBlock(height uint64) (*block.Block, error) {
 		return nil, db.ErrNotExist
 	}
 	// check whether block in staging buffer or not
-	storeKey := blockStoreKey(height, fd.header)
-	if storeKey >= fd.blkStore.Size() {
-		blkStore, err := fd.blkBuffer.Get(stagingKey(height, fd.header))
-		if err != nil {
-			return nil, err
-		}
+	if blkStore, err := fd.getFromStagingBuffer(height); err == nil {
```

Comment:
It is best to check for errors and throw if unexpected

Reply:
updated in 2nd commit